ts/Extensions/BasePopup.ts

```diff
@@ -0,0 +1,189 @@
+/* *
+ *
+ *  Base popup.
+ *
+ *  (c) 2009 - 2023 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import AST from '../Core/Renderer/HTML/AST.js';
+import U from '../Core/Utilities.js';
+const {
+    addEvent,
+    createElement
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+abstract class BasePopup {
```

Comment:
I remember we started the discussion about the name. How about `Form`/`BaseForm`? Then changing `formType` property to a simple `type`

Bringing @bre1470 and @sebastianbochan to the discussion :)

Reply:
Changed to `BaseForm` and `type`

Reply:
Sounds good!

Reply:
Unfortunately I cannot agree, the `BaseForm` means nothing for me. 

For me the `MockPopup` or `Popup` as a base sounds more intuitive. Then, we can extend the class by a specified form of popup like `AnnotationPopup`.

TL:LR;
I would craete an abstract class as `Popup` and extend it by a specific one like `AnnotationPopup` / `ConfirmationPopup` /  `DashboardPopup` / `DataPopup` etc.

Reply:
For now, we decided only to extract that abstract class. 
Creating separate classes for each file will not be part of the PR.

Regarding naming, I think `MockPopup` sounds like it is something for temporary use.
`BaseForm`/`BasePopup` works for me :)

Reply:
I agree with Karol, and I'm okay with both: `BasePopup` and `BaseForm` 👍

Reply:
Seeing all green marks.. I guess the decision is up to @karolkolodziej ;)